data/prepro.py

```diff
@@ -28,15 +29,21 @@ def tokenize_data(data, word_count=False):
         ans_toks.append(word_tokenize(i))
 
     for i in data['data']['dialogs']:
-        # pad i['dialog'] with fake question-answer pairs in the beginning
-        while len(i['dialog']) < 10:
-            i['dialog'].insert(0, {'question': -1, 'answer': -1})
         if 'answer' not in i['dialog'][-1]:
             i['dialog'][-1]['answer'] = -1
+        # pad i['dialog'] with fake question-answer pairs at the end
+        while len(i['dialog']) < 10:
+            i['dialog'].append({'question': -1, 'answer': -1})
         res[i['image_id']]['dialog'] = i['dialog']
         for j in range(10):
-            question = ques_toks[i['dialog'][j]['question']]
-            answer = ans_toks[i['dialog'][j]['answer']]
+            if i['dialog'][j]['question'] == -1:
+                question = ''
+            else:
+                question = ques_toks[i['dialog'][j]['question']]
+            if i['dialog'][j]['answer'] == -1:
+                answer = ''
+            else:
+                answer = ans_toks[i['dialog'][j]['answer']]
```

Comment:
These checks and conversions were not needed since this loop forms the word count -- word count is only formed with either `train` or `trainval` and not with `test`. So I removed these conditions and kept a check of `word_count == True` outside the loop. I think this is safe and reduces some clutter.